master/CMakeLists.txt

```diff
@@ -25,11 +25,14 @@ if(MONGOCLIENT_FOUND)
     list(APPEND master_plugins mongodb_assigner.cpp)
 endif(MONGOCLIENT_FOUND)
 
-set(husky husky-lib husky-io husky-core husky-base)
+add_library(husky-master-objs OBJECT master.cpp ${master_plugins})
+husky_default_properties(husky-master-objs)
+
+set(husky-master-dep husky-io husky-core husky-base)
 
 # Master
-add_executable(Master master.cpp ${master_plugins})
-target_link_libraries(Master ${husky})
+add_executable(Master master_main.cpp $<TARGET_OBJECTS:husky-master-objs>)
```

Comment:
The `husky_default_properties` is in the end of the file, please check.

The variable after .cpp are the objects that contain global variables which need to be initialized.